xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-image/plugin.js

```diff
@@ -100,13 +101,283 @@
+                extraClass: "ckeditor-autocomplete-centered",
+                space: "",
+                badge: "",
+                imgSrc: "",
+                type: "icon",
+              };
+
+            // Upload image item should be the first in the list
+            const topItems = [];
+            if (uploadItem.label.toLowerCase().startsWith(opts.query.toLowerCase())) {
+              topItems.push(uploadItem);
+            }
+
+
+            const isImage = function (attachment) {
+              return attachment.mimetype.some(function (mime) {
+                return mime.startsWith("image/");
+              });
+            };
+
+            callback(topItems.concat([loadingItem]));
+
+            // Start all the Solr Queries
+            const localQuery = attachmentService.getAttachments(`*${opts.query}*`, false, false, editor, "desc");
+            const globalQuery = attachmentService.getAttachments(`*${opts.query}*`, true, false, editor, "desc");
+
+            const wildcardLocalQuery = attachmentService.getAttachments("*", true, false, editor, "desc");
+            const wildcardGlobalQuery = attachmentService.getAttachments("*", true, false, editor, "desc");
+
+            // Display the wildcard query result first because it is more likely to be cached
+            const wildcardQuery = Promise.all(
+              // Display local results higher in the list
+              [wildcardLocalQuery, wildcardGlobalQuery]
+            ).then(function (values) {
+              // Show a loading entry until we show the result from the specific query
+              const wildcardItems = [...topItems, loadingItem];
+
+              // Store the attachments so we can display them once we get the specific query results
+              const wildcardAttachments = [];
+
+              // Keep track of the items already in the list to avoid duplicates
+              const wildcardIncludedIds = {};
+              values.flat().forEach(function (attachment) {
+                if (isImage(attachment) &&
+                  !wildcardIncludedIds[attachment.id] &&
+                  // Filter based on the filename since we're using a wildcard query
+                  attachment.filename[0].toLowerCase().includes(opts.query.toLowerCase())) {
+                  wildcardIncludedIds[attachment.id] = true;
+                  wildcardAttachments.push(attachment);
+                  wildcardItems.push(attachmentToItem(attachment));
+                }
+              });
+
+              // Show the wildcard query's results
+              callback(wildcardItems);
+
+              return wildcardAttachments;
+            });
+
+            // Display all the results once the solr requests are finished
+            Promise.all([
+              // Display the specific query results higher in the list
+              localQuery, globalQuery, wildcardQuery
+            ]).then(function (values) {
+
+              const finalItems = [...topItems];
```

Comment:
Note that is assignment is creating an entirely new "cloned" array. Very small perf improvement but you could just assign topItems as the inital array afaics (since it is not manipulated in the rest of the method)

Reply:
Indeed, I'll change it, thank you.